File: 122-BestTimetoBuyandSellStockII/122-BestTimetoBuyandSellStockII.java
// Last updated: 12/22/2025, 12:31:13 PM
1class Solution {
2    public int maxProfit(int[] prices) {
3        int profit = 0;
4        for(int i = 1; i< prices.length; i++){
5            if(prices[i] > prices[i-1]){
6                profit += prices[i] - prices[i-1];
7            }
8        }
9        return profit;
10    }
11}

File: 122-BestTimetoBuyandSellStockII/122-BestTimetoBuyandSellStockII.js
// Last updated: 1/20/2026, 11:02:42 AM
1/**
2 * @param {number[]} prices
3 * @return {number}
4 */
5var maxProfit = function(prices) {
6    let profit = 0;
7    for(let i = 1;i < prices.length; i++){
8        if(prices[i] > prices[i-1]){
9            profit += prices[i] - prices[i-1];
10        }
11    }
12    return profit;
13};